src/rulesets/reader.ts

```diff
@@ -4,15 +4,17 @@ import { extname, join } from '@stoplight/path';
 import { Optional } from '@stoplight/types';
 import { parse } from '@stoplight/yaml';
 import { readFile, readParsable } from '../fs/reader';
-import { httpAndFileResolver } from '../resolvers/http-and-file';
+import { createHttpAndFileResolver } from '../resolvers/http-and-file';
 import { FileRulesetSeverity, IRuleset, RulesetFunctionCollection } from '../types/ruleset';
 import { findFile, isNPMSource } from './finder';
 import { mergeFormats, mergeFunctions, mergeRules } from './mergers';
 import { mergeExceptions } from './mergers/exceptions';
 import { assertValidRuleset } from './validation';
+import { Agent } from 'http';
 
 export interface IRulesetReadOptions {
   timeout?: number;
+  agent?: Agent;
```

Comment:
@adanilev Shouldn't we rather extend from `IHttpAndFileResolverOptions` here?

Reply:
@nulltoken, makes sense. Updated!